apps/server/src/modules/h5p-editor/contentStorage/contentStorage.spec.ts

```diff
@@ -1,471 +1,905 @@
+	let contentRepo: DeepMocked<H5PContentRepo>;
 
 	beforeAll(async () => {
 		module = await Test.createTestingModule({
 			providers: [
-				{
-					provide: ContentStorage,
-					useValue: new ContentStorage(dir, { invalidCharactersRegexp: /!/ }),
-				},
+				ContentStorage,
+				{ provide: S3ClientAdapter, useValue: createMock<S3ClientAdapter>() },
+				{ provide: H5PContentRepo, useValue: createMock<H5PContentRepo>() },
 			],
 		}).compile();
+
 		service = module.get(ContentStorage);
+		s3ClientAdapter = module.get(S3ClientAdapter);
+		contentRepo = module.get(H5PContentRepo);
 	});
 
 	afterAll(async () => {
 		await module.close();
 	});
 
 	afterEach(() => {
-		rimraf.sync(dir);
 		jest.resetAllMocks();
 	});
 
-	it('should be defined', () => {
+	it('service should be defined', () => {
 		expect(service).toBeDefined();
 	});
 
 	describe('addContent', () => {
-		describe('WHEN content is created successfully', () => {
-			it('should return contentId', async () => {
-				const { metadata, content, user } = setup();
-				const contentID = await service.addContent(metadata, content, user);
-				expect(contentID).toBeDefined();
-				expect(typeof contentID).toBe('string');
-			});
-		});
-		describe('WHEN contentId is empty', () => {
-			it('should throw new error', async () => {
-				const { metadata, content, user, emptyContentId } = setup();
-				try {
-					await service.addContent(metadata, content, user, emptyContentId);
-				} catch (err) {
-					expect(err).toBeInstanceOf(Error);
-				}
+		const setup = () => {
+			const newContent = helpers.buildContent(0).new();
+			const existingContent = helpers.buildContent(0).withID();
+
+			const user = helpers.createUser();
+
+			return { newContent, existingContent, user };
+		};
+
+		describe('WHEN adding new content', () => {
+			it('should call H5pContentRepo.save', async () => {
+				const {
+					newContent: { metadata, content },
+					user,
+				} = setup();
+
+				await service.addContent(metadata, content, user);
+
+				expect(contentRepo.save).toHaveBeenCalledWith(expect.objectContaining({ metadata, content }));
+			});
+
+			it('should return content id', async () => {
+				const {
+					newContent: { metadata, content },
+					user,
+				} = setup();
+				contentRepo.save.mockImplementationOnce(helpers.repoSaveMock);
+
+				const id = await service.addContent(metadata, content, user);
+
+				expect(id).toBeDefined();
 			});
 		});
-		describe('WHEN content can not be added error', () => {
-			it('should throw new error', async () => {
-				const { metadata, content, user } = setup();
-				jest.spyOn(fsPromiseMock, 'writeFile').mockImplementationOnce(() => {
-					throw new Error('Could not write file');
-				});
-				try {
-					await service.addContent(metadata, content, user);
-				} catch (err) {
-					expect(err).toBeInstanceOf(Error);
-				}
+
+		describe('WHEN modifying existing content', () => {
+			it('should call H5pContentRepo.findById', async () => {
```

Comment:
Here is a topic I'm debating about with others for some time now: I don't think you need to test for this call because it has no side effects. IMO you only need to test for calls that have side effects like e.g. repo.save. 

Some reasoning: There is this point in our docs "stick to blackbox testing: think about the unit from the outside, not its inner workings" (https://hpi-schul-cloud.github.io/schulcloud-server/additional-documentation/nestjs-application/testing.html). So in our case here, if you would gather the needed data some other way than by calling findById, the test should not fail. It should not bother with how you do that, but just with the outcome. i.e. that save is called with the right arguments.

Unfortunately you can't really stick to black box testing if you have side effects, like we have here with the save call. But I think it is good to do it as much as possible.

Reply:
I would mostly agree with you, but findById actually does have some side effects, since the returned value is being modified and then passed over to .save (Not in this particular test, but the next one). 
But the next test already mocks the return value of findById, so I guess this test is kind of superfluous. I'll just remove it.